search/collector/topn.go

```diff
@@ -154,6 +154,23 @@ func (hc *TopNCollector) Collect(ctx context.Context, searcher search.Searcher,
 		hc.sort.UpdateVisitor(field, term)
 	}
 
+	var dmHandlerMaker search.MakeDocumentMatchHandler
+	var dmHandler search.DocumentMatchHandler
+	var loadID, ok bool
+	if cv := ctx.Value(search.MakeDocumentMatchHandlerKey); cv == nil {
+		// initialize the dmHandler with the default TopN handler
+		dmHandler =  NewTopNDocumentMatchHandler(searchContext, hc, nil)
```

Comment:
There's something not quite right about this, in that the NewTopNDocumentMatchHandler() func signature takes this extra hc param, so it doesn't have the same exact signature as the MakeDocumentMatchHandler func type.

That means, I think, these things are not chainable or wrappable.  Like, my application should be able to have its own handler, which might does some extra logging or stats or streaming or something, but then should be able to pass along the hit to a top-N handler.  But, my app can't construct a top-N handler if it doesn't have the hc instance?